### DISCUSSION TITLE
Two way data transport using REST API

### DISCUSSION THREAD

**Discussion by Drayang** [frePPLe/frepple#468] [Q&A]
From what I found out the REST API feature only allow the user to export data to a 3rd party software but it does not allow the user to export the data from 3rd party software to the frepple itself. 

Is that any method that allow the user to transport the data (in JSON form) from a 3rd part software (e.g an ERP system) to frepple itself via REST API?

Any help is appreciate! Thank you.

ps: Do not want to export or import using CSV or Excel file

**Comment by jdetaeye** [comment]
>>From what I found out the REST API feature only allow the user to export data to a 3rd party software but it does not allow the user to export the data from 3rd party software to the frepple itself.

That's absolutely incorrect!
The REST API is two-way.   You can read and write data.

>>ps: Do not want to export or import using CSV or Excel file

In practice we find that CSV/excel extracts are to be the simplest, most used, and most robust types of integration.
An application like frepple needs a snapshot of all planning related ERP at a certain moment.  That's a large and complex set of inter-related objects like inventory, open sales orders, open purchase orders, open manufacturing orders, etc.  Getting a consistent and complete picture is very important.  One should rather focus on getting that right, rather than then technical choice of csv, excel or rest.

**Reply by Drayang** [reply]
Thank you the reply. What i trying to do is to export the data from an ERP system to the frepple via REST API and automatically create the demand/buffer/etc. in frepple for production planning purpose. I am thinking by doing in this way I can have a real time data transport like how the Odoo Connector work (no very sure how the connector work but what I mean is to transport data directly from ERP software to frepple by POST or GET request). However, I am kinda unsure on how the data transport can be done via REST API on frepple side. Do you have any idea or comment on this?
